src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1413,7 +1416,112 @@ private IBlockOperation CreateBoundBlockOperation(BoundBlock boundBlock)
             ITypeSymbol type = null;
             Optional<object> constantValue = default(Optional<object>);
             bool isImplicit = boundBlock.WasCompilerGenerated;
-            return new CSharpLazyBlockOperation(this, boundBlock, locals, _semanticModel, syntax, type, constantValue, isImplicit);
+            var parameters = (_semanticModel as MemberSemanticModel)?.MemberSymbol.GetParameters() ?? ImmutableArray<ParameterSymbol>.Empty;
+            var nullChecksToPrepend = GenerateNullChecksForParameters(parameters, syntax);
```

Comment:
Sorry, my previous comment was confusing. I think it makes sense to check for `MemberSemanticModel` here but the check for `IsNullChecked` seemed unnecessary.

Perhaps:
```
var nullChecksToPrepend =
    (_semanticModel is MemberSemanticModel memberModel && blockInModel) ?
    GenerateNullChecksForParameters(memberModel.MemberSymbol.GetParameters()) :
    ImmutableArray<ConditionalOperation>.Empty;
```